cfgov/v1/models/handlers.py

```diff
@@ -0,0 +1,78 @@
+from collections import OrderedDict
+from itertools import chain
+from wagtail.wagtailcore import blocks
+from wagtail.wagtailcore.blocks.stream_block import StreamValue
+
+from ..atomic_elements import ATOMIC_JS
+
+
+class Handler(object):
+    def __init__(self, page, request):
+        self.page = page
+        self.request = request
+
+    def process(self, context):
+        raise NotImplementedError
+
+    # Retrieves the stream values on a page from it's Streamfield
+    def get_streamfield_blocks(self):
+        lst = [value for key, value in vars(self.page).iteritems()
+               if type(value) is StreamValue]
+        return list(chain(*lst))
+
+
+class JSHandler(Handler):
+    """
+        Gathers all the JS files specific to this page and its current
+        Streamfield's blocks and puts them in the template context.
+    """
+    def __init__(self, page, request):
+        super(JSHandler, self).__init__(page, request)
+        self.js_dict = OrderedDict()
+
+    def process(self, context):
+        self.generate_js_dict()
+        context['media'] = self.js_dict
```

Comment:
In this PR there's only a single `Handler` in [this](https://github.com/cfpb/cfgov-refresh/pull/2301/files#diff-8e8a3e28d46ee12516bea4f5b50bf5a4R222) loop. If there are multiple handlers in future, won't assigning `context['media']` like this potentially overwrite some other media that another handler might have put in the dictionary?